proposals/unsafe-evolution.md

```diff
@@ -0,0 +1,138 @@
+[pointer element access][pointer-element-access], should be worked into [§12](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/expressions.md), with references to `unsafe` contexts
+removed. No semantics change about the meaning of these expressions; the only change is that they no longer require an `unsafe` context to use.
+
+For [pointer indirection][pointer-indirection], [pointer member access][pointer-member-access], and [pointer element access][pointer-element-access], these operators remain unsafe, as these
+access memory that is not managed the runtime. They remain in [§24][unsafe-code.md], and continue to require an `unsafe` context to be used. Any use outside of an `unsafe` context is an error.
+No semantics about these operators change; they still continue to mean exactly the same thing that they do today.
+
+The [fixed statement][fixed-statement] moves to [§13](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/statements.md), with references to `unsafe` contexts removed.
+
+#### Fixed-size buffers
+
+The story for [fixed-size buffers][fixed-size-buffer-declarations] is similar to [pointers](#pointer-types). The definition of a fixed-size buffer is not itself dangerous, and moves to
+[§16.3](https://github.com/dotnet/csharpstandard/blob/draft-v8/standard/structs.md#163-struct-members). Accessing a fixed-size buffer in an expression is similarly safe, unless the expression occurs
+as the _primary_expression_ of an `element_access`; these are evaluated as a _pointer_element_access_, which is unsafe, as per the rules above.
+
+#### Stack allocation
+
+Again, the story for [stack allocation][stack-allocation-spec] is very similar to [pointers](#pointer-types). Converting a `stackalloc` to a pointer is no longer unsafe; it is the deference of that
+pointer that is unsafe. We do add one new rule, however:
+
+A _stackalloc_expression_ has a [safety state](#expression-safety-state) of unsafe if all of the following statements are true:
+
+* The _stackalloc_expression_ is being converted to a `Span<T>` or a `ReadOnlySpan<T>`.
+* The _stackalloc_expression_ does not have a _stackalloc_initializer_.
+* The _stackalloc_expression_ is used within a member that has `SkipLocalsInitAttribute` applied.
+
+In these contexts, the resulting stack space could have unknown memory contents, and it is being converted to a type that provides a safe wrapper around unmanaged memory access. This violates the
+contract of `Span<T>` and `ReadOnlySpan<T>`, and so must be subject to extra scrutiny by the author and reviewers of such code.
+
+> [!NOTE]
+> This means that assigning a `stackalloc` to a pointer is _always_ safe, regardless of context. Arguably, this is a bit of failure of the proposal, as the creation of the pointer could be far from
+> the actual dereference of the pointer.
```

Comment:
This note doesn't seem specific to `stackalloc`. Indeed, obtaining or manipulating pointers in an unsafe way could be done in a different place from the de-reference.